### PR TITLE
Update release workflow to push container to ECR of Marketplace

### PR DESCRIPTION
**Pull request by kota2and3kan**
This PR updates release workflow to push container image to ECR of AWS Marketplace.
I use Identity Provider (OpenID Connect) instead of access key and secret access key to authenticate ECR in the CI.
I will add some comment in the code.

Please take a look!

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -5,6 +5,10 @@ on:
     tags:
       - "v[0-9]+.[0-9]+.[0-9]+"
 
+permissions:
+  id-token: write
+  contents: read
```
Comment (kota2and3kan) [inline .github/workflows/release.yaml:10]: To get `ID token` from GitHub OIDC Provider, we need to add permission configuration (allow) `id-token: write`.
https://docs.github.com/ja/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-cloud-providers

Also, if we add `permissions` in the YAML file, the default permission `contents: read` is NOT allowed.
So, in this case, we need to set (allow) `contents: read` explicitly.
The `contents: read` is required to checkout this repository in the CI (it's required for `actions/checkout@v2`).

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -51,6 +55,17 @@ jobs:
           username: ${{ secrets.ACR_USER }}
           password: ${{ secrets.ACR_PASSWORD }}
 
+      - name: Configure AWS credentials
```
Comment (kota2and3kan) [inline .github/workflows/release.yaml:58]: This step gets credentials from AWS to login ECR using OpenID Connect.

File: .github/workflows/release.yaml [hunk 3]
```diff
@@ -51,6 +55,17 @@ jobs:
...
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          role-to-assume: ${{ secrets.AWS_MARKETPLACE_ROLE_OIDC }}
+          aws-region: us-east-1
+
+      - name: Login to Amazon Elastic Container Registry
```
Comment (kota2and3kan) [inline .github/workflows/release.yaml:64]: This step login to ECR of AWS Marketplace using credentials that we got in the above step.

File: .github/workflows/release.yaml [hunk 4]
```diff
@@ -66,12 +81,19 @@ jobs:
...
 
+      # Push the latest tag image since it is required from the perspective of the specification of Azure Marketplace.
       - name: Push latest tag to Azure Container Registry
         if: ${{ steps.version.outputs.version == steps.version.outputs.latest_version }}
         run: |
           docker tag ghcr.io/scalar-labs/scalardb-server:${{ steps.version.outputs.version }} ${{ secrets.ACR_LOGIN_SERVER }}/scalardb-server:latest
           docker push ${{ secrets.ACR_LOGIN_SERVER }}/scalardb-server:latest
 
+      # Push the patch version image only since AWS Marketplace restricts using the latest tag.
+      - name: Push containers to Amazon Elastic Container Registry
```
Comment (kota2and3kan) [inline .github/workflows/release.yaml:92]: We push the patch version image only since AWS Marketplace restricts using the latest tag.

**Comment by kota2and3kan** [comment]
CI is failed since the Trivy vulnerability test is failed.

**Review by Torch3333 — APPROVED** [review]
LGTM, thank you!

**Review by brfrn169 — APPROVED** [review]
LGTM! Thank you!

I realized that we should not have add such company things originally as this is published as OSS. Let's discuss it later. (CC: @feeblefakie)

**Comment by kota2and3kan** [comment]
@brfrn169  
Thank you for your review!
Indeed, we should take care of the license. 

I think we can separate the container pushing part from this repository's CI to another private repository's CI. Let's discuss it later!

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**